nucypher/utilities/nodedeploy.py

```diff
@@ -0,0 +1,426 @@
+import os
+import json
+import maya
+import time
+import hashlib
+from base64 import b64encode
+
+from jinja2 import Template, TemplateError
+
+from botocore import exceptions as botoexceptions
+
+from nucypher.config.constants import DEFAULT_CONFIG_ROOT
+from nucypher.blockchain.eth.clients import PUBLIC_CHAINS
+from nucypher.blockchain.eth.networks import NetworksInventory
+from nucypher.blockchain.eth.registry import BaseContractRegistry
+
+NODE_CONFIG_STORAGE_KEY = 'worker-configs'
+URSULA_PORT = 9151
+PROMETHEUS_PORTS = [9101, 9090]
+
+def preflight_check(*attrs):
+    necessary_attrs = attrs
+    def wrapper(fn):
+        def wrapped(self, *args, **kwargs):
+            if all([getattr(self, a, False) for a in necessary_attrs]):
+                return fn(self, *args, **kwargs)
+        return wrapped
+    return wrapper
+
+
+class CloudNodeConfigurator:
+    """
+    configures an existing running node
+    """
+
+    clef_rules_template = """
+
+function ApproveListing() {
+    return "Approve"
+}
+
+function ApproveTx(r) {
+    if (req.transaction.to.toLowerCase() == "{{confirm_activity_address}}") {
+        return "Approve"
+    }
+}
+
+function ApproveSignData(r) {
+    return "Approve"
+}
+
+"""
+    @property
+    def chain_id(self):
+        return NetworksInventory.get_ethereum_chain_id(self.network)
+
+    @property
+    def chain_name(self):
+        try:
+            return PUBLIC_CHAINS[self.chain_id].lower()
+        except KeyError:
+            self.emitter.echo(f"could not identify public blockchain for {self.network}", color="red")
+
+    def get_signer_path(self, clef):
+        if clef:
+            return '/root/.local/share/clef/Signer/clef.ipc'
+        return None
+
+
+
+    def generate_ansible_inventory(self, blockchain_provider=None, clef=False):
+
+        status_template = Template(self.inventory_template)
+
+        if clef:
```

Comment:
There is clearly some thinking done here... Unfortunately - I think that clef is a bad fit for this style of deployment and I think this PR will be more focused without it.

Reply:
Yeah.  I just did so much work on it, I didn’t want to just remove it!  What if it suddenly becomes useful?

Reply:
I hear ya.  In an effort to sift through the immediate and ongoing needs, we dont need to abandon the clef changes completely.  Perhaps remove them from this PR, then open another Draft based on this one that extends the basic functionality.